### PR TITLE
Update notebooks with current API

### LINKED ISSUE #1444: Add documentation on running tests and building docs
Just want to comment here that I'm working on making sure example notebooks run smoothly.  I'm not familiar with a best practice for notebooks in a repo -- let me know if there are specifics with how to run them before committing (i.e., should stderr be suppressed?)

### PR DESCRIPTION
**Pull request by ColCarroll**
Mostly making sure imports still worked, restarting kernels and running the code all the way through.  Did not check much for accuracy or soundness of approaches.  

The only substantive change was in `trace_to_dataframe`, which now includes transformed variables.  One of the notebooks requires this to run, and it made sense to me that a user can filter columns more easily than recreating the dropped ones, but I can revert that if there are objections.

This was all run in python 3.5, using the environment we build on travis, with a few extra libraries installed manually if the notebook required it.  

Fixes #1444 , except for `pmf-pymc`, which has a separate issue (#830).

### CODE REVIEW

File: docs/source/notebooks/profiling.ipynb [hunk 1]
```diff
@@ -12,28 +12,33 @@
...
+    "returns = np.genfromtxt(get_data_file('pymc3.examples', \"data/SP500.csv\"))[-n:]\n",
+    "with Model() as model:\n",
     "    sigma = Exponential('sigma', 1. / .02, testval=.1)\n",
-    "\n",
     "    nu = Exponential('nu', 1. / 10)\n",
-    "\n",
     "    s = GaussianRandomWalk('s', sigma ** -2, shape=n)\n",
-    "\n",
-    "    r = StudentT('r', nu, lam=exp(-2 * s), observed=returns)\n",
+    "    r = StudentT('r', nu, lam=np.exp(-2 * s), observed=returns)\n",
```
Comment (twiecki) [inline docs/source/notebooks/profiling.ipynb]: Shouldn't that be `pymc3.math.exp`?

File: pymc3/backends/tracetab.py [hunk 2]
```diff
@@ -21,16 +21,13 @@ def trace_to_dataframe(trace, chains=None, flat_names=None):
     """
     var_shapes = trace._straces[0].var_shapes
     if flat_names is None:
-        flat_names = {v: create_flat_names(v, shape)
-                      for v, shape in var_shapes.items()
-                      if not v.endswith('_')}
+        flat_names = {v: create_flat_names(v, shape) for v, shape in var_shapes.items()}
```
Comment (twiecki) [inline pymc3/backends/tracetab.py]: Is this reverting https://github.com/pymc-devs/pymc3/pull/1367? Why?

Comment (twiecki) [reply pymc3/backends/tracetab.py]: Ah, you mentioned that. Perhaps we can add a `kwarg` like `hide_transformed_vars`?

File: pymc3/backends/tracetab.py [hunk 3]
```diff
@@ -23,11 +23,11 @@ def trace_to_dataframe(trace, chains=None, flat_names=None):
     if flat_names is None:
         flat_names = {v: create_flat_names(v, shape)
                       for v, shape in var_shapes.items()
-                      if not v.endswith('_')}
+                      if not hide_transformed_vars or not v.endswith('_')}
```
Comment (twiecki) [inline pymc3/backends/tracetab.py]: equivalent to the more readable: `if not (hide_transformed_vars and v.endswith('_'))`

Comment (twiecki) [reply pymc3/backends/tracetab.py]: @ColCarroll If you changes this we can merge this.

**Comment by ColCarroll** [comment]
Missed that one, sorry!

**Comment by twiecki** [comment]
:+1:

**Pull request merged**